wps_tools/output_handling.py

```diff
@@ -0,0 +1,181 @@
+import json, requests, math
+
+from netCDF4 import Dataset
+from tempfile import NamedTemporaryFile
+from bs4 import BeautifulSoup
+from urllib.request import urlopen, urlretrieve
+from rpy2 import robjects
+
+from wps_tools.file_handling import copy_http_content
+from wps_tools.R import load_rdata_to_python, get_package
+
+
+def nc_to_dataset(url):
+    """
+    Access content of a netcdf file from an http url as a Dataset object
+    using the netCDF4 library.
+
+    Parameters:
+        url (str): http url path to a netCDF file
+
+    Returns:
+        Dataset: Dataset object containing input netCDF file content
+    """
+
+    with NamedTemporaryFile(
+        suffix=".nc", prefix="tmp_copy", dir="/tmp", delete=True
+    ) as tmp_file:
+
+        return Dataset(copy_http_content(url, tmp_file))
+
+
+def json_to_dict(url):
+    """
+    Access content from a json url file as a Python dictionary
+
+    Parameters:
+        url (str): file or http url path to a json file
+
+    Returns:
+        dictionary: Python dictionary with input json file's content
+    """
+
+    with NamedTemporaryFile(
+        suffix=".json", prefix="tmp_copy", dir="/tmp", delete=True
+    ) as json_file:
+        urlretrieve(url, json_file.name)
+
+        return json.load(json_file)
```

Comment:
There is plenty of similarity between all the methods here. That being said I think you have distilled it as far as it can go, nice work!